xdsl/ir/core.py

```diff
@@ -583,6 +587,32 @@ def remove_use(self, use: Use):
         assert use in self._uses, "use to be removed was not in use list"
         self._uses.remove(use)
 
+    def has_single_use(self) -> bool:
+        """Returns true if the value has exactly one use."""
+        return len(self._uses) == 1
+
+    def get_single_use(self) -> Use | None:
+        """
+        Returns the single use of the value, or None if there are no uses or
+        more than one use.
+        """
+        if len(self._uses) == 1:
+            return next(iter(self._uses))
+        return None
+
+    def get_single_use_user(self) -> Operation | None:
+        """
+        Returns the user of the single use of the value.
+        If there are no uses or more than one use, returns None.
+        """
+        if len(self._uses) == 1:
+            return next(iter(self._uses)).operation
```

Comment:
```suggestion
        if (use := self.get_single_use()) is not None:
            return use.operation
```
Tbh this helper function seems of questionable utility to me.

Reply:
(I guess this code gets nuked later though)

Reply:
Yeah I would not implement it either way once the implementation is changed

Reply:
I changed it, but yeah it's getting nuked in 2 commits anyway